securedrop/tests/client_server_test.py

```diff
@@ -86,18 +103,40 @@ def test_aad_initial_ask_to_register_password_too_short(self):
                     with self.assertRaises(RuntimeError):
                         client.main()
 
-    def test_aad_initial_registration_succeeds(self):
+    def test_aaf_initial_registration_succeeds(self):
+        """Ensures that client doesn't throw during some wild, yet valid user registrations."""
+        valid_emails = ["simple@example.com", "very.common@example.com",
+                        "x@example.com",
+                        "user%example.com@example.org",
+                        "mailhost!username@example.org",
+                        "er.name+tag+sorting@example.com",
+                        "fully-qualified-domain@example.com",
+                        "example-indeed@strange-example.com",
+                        "other.email-with-hyphen@example.com",
+                        "EXTREMELYLONGEMAIL12345678901234567890123456789012345678901234+x@example.com"
+                        ]
+        with server_process():
+            for i in valid_emails:
+                se1 = InputSideEffect(["y", "name_v", i, "exit"])
+                se2 = InputSideEffect(["password_v12", "password_v12"])
+                with patch('builtins.input', side_effect=se1.se):
+                    with patch('getpass.getpass', side_effect=se2.se):
+                        client.main()
+                        os.remove("client.json")
```

Comment:
Reference filename variables in server/client.py directly instead of using string literals.

Reply:
Good Idea.